qz-print/src/qz/PrintHTML.java

```
@@ -86,34 +86,42 @@ public String get() {
         JFrame j = new JFrame(jobName.get());
         j.setUndecorated(true);
         j.setLayout(new FlowLayout());
         this.setBorder(null);
         
-        for (String s : getHTMLDataArray()) {
-            this.setText(s + "</html>");
-            j.add(this);
+        try{
+            for (String s : getHTMLDataArray()) {
+                this.setText(s + "</html>");
+                j.add(this);
 
 
-            j.pack();
-            j.setExtendedState(j.ICONIFIED);
-            j.setVisible(true);
+                j.pack();
+                j.setExtendedState(j.ICONIFIED);
+                j.setVisible(true);
 
-            // Elimate any margins
-            HashPrintRequestAttributeSet attr = new HashPrintRequestAttributeSet();             
-            attr.add(new MediaPrintableArea(0f, 0f, getWidth()/PrintPostScript.DPI, getHeight()/PrintPostScript.DPI, MediaPrintableArea.INCH));               
+                // Elimate any margins
+                HashPrintRequestAttributeSet attr = new HashPrintRequestAttributeSet();             
+                attr.add(new MediaPrintableArea(0f, 0f, getWidth()/PrintPostScript.DPI, getHeight()/PrintPostScript.DPI, MediaPrintableArea.INCH));               
 
-            PrinterJob job = PrinterJob.getPrinterJob();    
-            job.setPrintService(ps.get());
-            job.setPrintable(this);
-            job.setJobName(jobName.get());
-            job.print(attr);
-            j.setVisible(false);
+                PrinterJob job = PrinterJob.getPrinterJob();    
+                job.setPrintService(ps.get());
+                job.setPrintable(this);
+                job.setJobName(jobName.get());
+                job.print(attr);
+                j.setVisible(false);
+            }
+            j.dispose();
+            clear();
+            } 
+        catch(Exception e){
+            j.dispose();
+            clear();
+            
+            throw new PrinterException(e.toString());
```

You should be able to just throw e rather than instantiating a potentially unrelated PrinterException.